servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/AsyncContextProvider.java

```diff
@@ -47,6 +47,12 @@ interface AsyncContextProvider {
      */
     ContextMap context();
 
+    /**
+     * Set the current {@link ContextMap}
+     * @param contextMap the {@link ContextMap} to set in the thread local state.
+     */
+    void setContextMap(ContextMap contextMap);
+
     /**
      * Capture existing context in preparation for an asynchronous thread jump.
```

Comment:
Without going too much into the changes, could you show an example here how this approach will evolve into reusing the same abstraction for `captureContext/attachContext/ContextPreservingRunnable`?

Reply:
Answered in https://github.com/apple/servicetalk/pull/3183/commits/d73d0c0634cfd753b5c5f83aa05e746dd856a2bf#diff-6444304586f6a7960d00326c81cefa6355d889c1c8373c24b9be15c3acaedee1